examples/recording_service/pluggable_storage/c/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project (PluggableStorage)
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+
+find_package(
+    RTIConnextDDS "6.0.0"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+find_package(
+    RTICodeGenerator "3.0.0"
+    REQUIRED
+)
+
+# Generate type plug-in files for the storage type
+add_custom_command(
+    OUTPUT 
+        ${CMAKE_SOURCE_DIR}/HelloMsg.c 
+        ${CMAKE_SOURCE_DIR}/HelloMsgPlugin.c
+        ${CMAKE_SOURCE_DIR}/HelloMsgSupport.c
+    MAIN_DEPENDENCY ${CMAKE_SOURCE_DIR}/HelloMsg.idl
+    COMMAND ${RTICODEGEN} -language C -replace -d ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/HelloMsg.idl
```

Comment:
It's the first time I see the MAIN_DEPENDENCY in the examples. Iis this they way to call rtiddsgen or do we have some other way/convention? @iblancasa

Reply:
We were not using it because it has some side effects. It is almost like "DEPENDS" but will do more things for VS.


Actually, something to improve would be:
* Create a custom target depending on the output files generated by the custom command  
* The libraries and the applications should depends on that custom

That will ensure Codegen is not called twice in multi-job builds (if Codegen is called twice, the build will break)

Reply:
Okay, so I'll create the custom target approach. What about the MAIN_DEPENDENCY, should I change it to DEPENDS?

Reply:
I don't have a strong opinion regarding  the ``MAIN_DEPENDENCY`` option.

Reply:
I don't have an strong opinion either, but it looks, that when there is a 1-1 dependency relationship (e.g. file generation), MAIN_DEPENDENCY might be better:

https://rix0r.nl/blog/2015/08/13/cmake-guide/